core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -175,8 +184,8 @@ class SnappyParser(session: SnappySession)
 
   protected final def numericLiteral: Rule1[Expression] = rule {
     capture(plusOrMinus.? ~ Consts.numeric. + ~ (Consts.exponent ~
-        plusOrMinus.? ~ CharPredicate.Digit. +).? ~ Consts.numericSuffix.?) ~
-        delimiter ~> ((s: String) => toNumericLiteral(s))
+        plusOrMinus.? ~ CharPredicate.Digit. +).? ~ Consts.numericSuffix.? ~
+        Consts.numericSuffix.?) ~ delimiter ~> ((s: String) => toNumericLiteral(s))
```

Comment:
This will cause suffixes like "bl", "df" to also be accepted (with last char used to determine) which is not proper.

Reply:
yes. that is true, but how to avoid that in parsing phase, i could not figure out. I am hoping that at later stage in analysis phase exception will be thrown.

Reply:
Either add checks for valid suffixes for each of the cases like mentioned below. Or you could use: ~ (ignoreCase("BD") | Consts.numericSuffix).? instead of two Consts.numericSuffix (and remove 'B' from numericSuffix). I think the first approach will be more efficient so just add checks for valid suffixes like mentioned in the comment below.